Source/WEB_ROOT/Inventory/js/editAsset.js

```diff
@@ -0,0 +1,206 @@
+
+$j(document).ready(function(){
+	$j('#deleteButton').click(function(e){
+		e.preventDefault();
+		$j('#deleteForm').submit();
+	});
+	$j('#btnSubmit').click(function(e){
+	    addRecordToHistoryDB("~(gpv.assetid)","Asset was Edited from the Edit Asset page by ~[x:username]");
+	});
+});
+
+
+// OnChange Events
+$j('#schoolDropDown').change(function(){
+		//The school drop down has changed
+		SetDropDowns();
+        $j('#storeSchoolId').val($j('#schoolDropDown :selected').val());
+        $j('#storeSchoolName').val($j('#schoolDropDown :selected').text());
+
+        //show the student / teacher check box
+        showHideFormElements("assignedToElement", true);
+});
+
+function createlisteners()
+{
+	$j('#teachersDropdown').change(function(){
+	    //The Teachers drop down has changed
+		//console.log($j('#teachersDropdown :selected').val());
+		 $j('#AssignedTo_dcid').val($j('#teachersDropdown :selected').val());
+		 $j('#AssignedTo_name').val($j('#teachersDropdown :selected').text());
+		
+		 if($j('#teachersDropdown :selected').val() != "")
+		 {
+			toggleAssetElements(true);
+		 }else{
+			toggleAssetElements(false);
+		 }
+		 
+	});
+
+	$j('#studentsToDropdown').change(function(){
+	    //The Teachers drop down has changed
+		//console.log($j('#studentsToDropdown :selected').val());
+		$j('#AssignedTo_dcid').val($j('#studentsToDropdown :selected').val());
+		$j('#AssignedTo_name').val($j('#studentsToDropdown :selected').text());
+		if($j('#studentsToDropdown :selected').val() != "")
+		 {
+			toggleAssetElements(true);
+		 }else{
+			toggleAssetElements(false);
+		 }
+	});
+}
+//End Onchange Events
+
+var assignedType;
+
+function SetDropDowns(){
+	//clear the dropdowns
+	document.getElementById('studentsToDropdown').options.length = 0;
+	document.getElementById('teachersDropdown').options.length = 0;
+
+	//run Ajax to get the dcid
+	 
+	if (document.getElementById('isStudentCheckBox').checked) 
+	{
+		assignedType = 1;
+	} else {
+		assignedType = 0;
+	};
+
+	var url = "scripts/AssignedToDropDown.html?type=" + assignedType +"&id=" + "&schoolid=" + $j('#schoolDropDown :selected').val();
+	//console.log(url);
+	$j.ajax(
+	{
+		url: url, 
+		dataType: "text",
+		success: function(result)
+		{
+			
+			if (assignedType == 1) 
+			{
+				//push students to the dropdown
+				var options = '<select id="studentsToDropdown">' + result + '</select>';
+				document.getElementById( 'studentsToDropdown' ).outerHTML = options;
+			} else {
+				//push teachers to the dropdown
+				var options = '<select id="teachersDropdown">' + result + '</select>';
+				document.getElementById( 'teachersDropdown' ).outerHTML = options;
+			};
+			createlisteners();

```

Comment:
This looks pretty solid John. I would say if you find yourself daisy chaining more than one other method within another method consider creating an separated function. That one function may be called something like InitializeForm(). Within that function you could call setdropdowns, createlisteners, ect ect. It would be really clear that the method manages the methods that do the actual work.